src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4246,6 +4246,64 @@ FCMP(double, d);
 
 #undef FCMP
 
+// Round double with mode
+
+void MacroAssembler::round_double_mode(FloatRegister dst, FloatRegister src, enum Round_double_mode round_mode, Register converted_dbl, Register mask, Register converted_dbl_masked)
+{
+
+  assert_different_registers(dst, src);
+  assert_different_registers(converted_dbl, mask, converted_dbl_masked);
+
+  // setting roundig mode to double->long (rm_direct) and long->double (rm_back) conversions
+  RoundingMode rm_direct, rm_back;
```

Comment:
Can we use the same rounding mode for conversions in both direction? Say `rup` for `ceil`, and `rdn` for `floor`.
I see this policy is used for both glibc [1] and V8. 

[1] https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/riscv/rv64/rvd/s_ceil.c;h=6c355cd72691c45c97201fe8947683287982ade9;hb=41d8c3bc33bcae1ebb8077b0442caef4917f763a

Reply:
basically for this case the backward rounding mode doesn't matter at all. it would only matter if intermediate integer contained something not representable in double.